Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -68,6 +68,30 @@ public class EntityProperties
         public Dictionary<Type, ComponentProperties> FromPrefab = new Dictionary<Type, ComponentProperties>();
 
         public Dictionary<Type, ComponentProperties> FromSpawn = new Dictionary<Type, ComponentProperties>();
+
+        private static object Get(Dictionary<Type, ComponentProperties> properties, Type componentType, string propertyName)
+        {
+            if (properties.ContainsKey(componentType) &&
+                properties[componentType].Values.ContainsKey(propertyName))
+            {
+                return properties[componentType].Values[propertyName];
+            }
+
+            return null;
+        }
+
+        public object Get<TComponent>(string propertyName, object defaultValue = null)
+        {
+            object result;
+
+            result = Get(FromSpawn, typeof(TComponent), propertyName);
+            if (result != null) { return result; }
+
+            result = Get(FromPrefab, typeof(TComponent), propertyName);
+            if (result != null) { return result; }
+
+            return defaultValue;
```

Comment:
This seems like a potential source of nasty bugs to me. Maybe I'm being idealistic, but I think we should yell and fail fast if we ask for a property that doesn't exist. Either that, or we could rename the method to `MaybeGet` to make it clear at the call site that it might return `null`.